src/pineko/theory.py

```diff
@@ -380,18 +381,29 @@ def fk(self, name, grid_path, tcard, pdf):
         max_al = 0
         # check for sv
         if not np.isclose(xir, 1.0):
-            is_ren_as, is_ren_al = check.contains_ren(grid, max_as, max_al)
-            if not (is_ren_as and is_ren_al):
-                raise ValueError(
-                    "Renormalization scale variations are not available for this grid"
-                )
+            checkres, max_as_effective = check.contains_sv(
```

Comment:
I know this function is more or less the central one in Pineko, but it is growing more and more, and start being too long and complex.

We should try to break it down in more atomic pieces.